### PR TITLE
Fix for generateThroughJoin

### PR DESCRIPTION
**Pull request by harshithkashyap**
### Pull Request check-list
- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does your issue contain a link to existing issue (Test Case for #[6052]).
- [x] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Have you added an entry under `Future` in the changelog?
### Description of change

Added Failing test case & patch for #6052

### CODE REVIEW

File: test/integration/include/findAll.test.js [hunk 1]
```diff
@@ -1961,5 +1961,92 @@ describe(Support.getTestDialectTeaser('Include'), function() {
...
+                  where: {
+                    tag_name: ['tag1', 'tag2']
+                  }
+                }
+              }]
+            }],
+            limit: 5,
+            offset: 0,
+            order: [
+              [ { raw: '\`Entity.votes\`' }, 'DESC' ]
```
Comment (mickhansen) [inline test/integration/include/findAll.test.js]: Please use proper order syntax for tests, we don't garantee `raw`

File: test/integration/include/findAll.test.js [hunk 2]
```diff
@@ -1961,5 +1961,92 @@ describe(Support.getTestDialectTeaser('Include'), function() {
...
+            ]
+        }).then(function (posts) {
+          expect(posts.length).to.equal(0);
```
Comment (mickhansen) [inline test/integration/include/findAll.test.js]: Best to verify that there is some results, and some results filtered.

**Comment by mickhansen** [comment]
Might be simpler to provide this as a sql unit test focusing on where the error happens.

File: test/integration/include/findAll.test.js (hunk 1 above)
Comment (harshithkashyap) [inline test/integration/include/findAll.test.js]: @mickhansen I have to use raw / sequelize.literal as there is problem with when using
```js
order: [
    [ this.sequelize.col('Entity.votes'), 'DESC' ]
]
```
I'll raise a separate issue for the same.

Comment (mickhansen) [inline test/integration/include/findAll.test.js]: @harshithkashyap And what when you use the proper nested order syntax from the docs?

Comment (harshithkashyap) [inline test/integration/include/findAll.test.js]: @mickhansen I did. I used
```js
order: [Entity,'votes', 'DESC']
```
1. Sequelize generates ``` ORDER BY `Entity`.`votes` DESC; ``` which should be ``` `Entity.votes` DESC ```. Here,  `Entity.votes` should be a single string
2. The order is not applied to the subquery

**Comment by harshithkashyap** [comment]
@mickhansen I've added insert queries & filtered expectations to the test. 
I tried looking into writing SQL unit test but it was a little confusing. This is a specific case when subQuery is thrown, there is an inner join with m:n relation using the through object in the join. Not sure how to generate all the cases. Could you guide me a little? Would be happy to submit the unit test.

**Comment by harshithkashyap** [comment]
@felixfbecker @sushantdhiman Unit Tests were requested by @mickhansen for this PR. But the `generateThroughJoin` method [here](https://github.com/sequelize/sequelize/blob/master/lib/dialects/abstract/query-generator.js#L1317) goes through a lot of pre-processing before being called and its very difficult to unit test the same. I've already added integration test for the same.

Incorrent Join identifier is used [here](https://github.com/sequelize/sequelize/blob/master/lib/dialects/abstract/query-generator.js#L1452) and i've submitted a patch for the same. Tests pass in my local instance. Travis CI seems to be stuck because of Deadlock. Could you guys restart the same?

**Comment by harshithkashyap** [comment]
@felixfbecker @sushantdhiman @janmeier @mickhansen Hey guys, could you review this please?

**Comment by felixfbecker** [comment]
I am not the right person to approve/reject these changes besides code style

**Comment by harshithkashyap** [comment]
Ping @janmeier

**Comment by mickhansen** [comment]
We have plenty of other unit tests for SQL generation for joins, it should be possible to cover this case aswell.

**Pull request merged**